src/simulations/simulator.py

```diff
@@ -146,7 +147,32 @@ def _extend_histogram(self, histogram, max_freq):
       return histogram + [0] * (max_freq - len(histogram))
     else:
       return histogram[:max_freq] 
-      
+
+  def _shuffle_distance(self, histogram1, histogram2):
+    """Computes shuffle distance between two histograms.
+
+    Given two frequency distributions f_1 and f_2, then shuffle distance
+    between them is defined as 
+
+       1/2 * sum_{k=1}^f |f_1(k) - f_2(k)|
+
+    Args:
+      histogram1:  A frequency histogram.  histogram1[k] is the number
+        of id's having a frequency of k+1 or greater.
+      histogram2:  A frequency histogram.  histogram2[k] is the number
+        of id's having a frequency of k+1 or greater.
+    Returns:
+      The shuffle distance between histogram1 and histogram2.
+    """
+    if len(histogram1) == 0 or len(histogram2) == 0:
+      return 0.0
+    counts1 = [histogram1[i] - histogram1[i+1] for i in range(len(histogram1)-1)] + [histogram1[-1]]
+    counts2 = [histogram2[i] - histogram2[i+1] for i in range(len(histogram2)-1)] + [histogram2[-1]]
+    max_freq = max(len(counts1), len(counts2))
+    freq1 = np.array(self._extend_histogram(counts1, max_freq)) / sum(counts1)
```

Comment:
this will never be that big, but np.sum would be faster in this function

Reply:
Done.